src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -646,10 +662,49 @@ export default function QuestionnaireEditor({ id }: QuestionnaireEditorProps) {
     return !hasError;
   };
 
+  const findFirstGroupErrorPath = (
```

Comment:
Shouldn't need all this custom logic either :)

We already have validateQuestions which calls findFirstErrorPath below. You just need to tweak it slightly to get it to scroll for group type errors.